avidemux_plugins/ADM_videoFilters6/mSharpen/ADM_vidMSharpen.cpp

```diff
@@ -118,7 +119,9 @@ ADMImage *src,*blur,*dst;
 
     dst->Pts=src->Pts;
 	
-    for (int i=0;i<3;i++)
+    invstrength=255-_param.strength;	// bugfix
```

Comment:
Would you please help me to understand why this is necessary?

In other words, in which circumstances can parameters be modified after object is constructed and getNextFrame() is called? I always thought that the current design tears down the filter chain and creates it anew at output start, but I can easily be in error.

Reply:
The bug can be reproduced the following way:
-load a video, go filters, select msharpen
-enter value (for example 255) to strength, then click OK
-click preview in the Video Filter Manager, and the result will be crap
-now enter the instantiated msharpen filter, and do nothing but click OK
-now the preview will be okay

Reply:
It looks like it only affects the preview of Video Filter Manager.
Is i understand it now, the invstrength only get value at object construction; leaving the configuration UI, the same object stay while in Video Filter Manager.

Reply:
Yes, there is a bug somewhere in filter chain code. When a filter is already present in the list of active filters, opening and accepting its configuration dialog creates a new instance while the old one is deleted => parameters are refreshed. This doesn't happen when a filter is initiallly added to the list.

Reply:
Well, the only thing which might be a bug is rebuilding filter chain when only the last filter was modified. You are absolutely right – we must be able to deal with configuration refresh between filter construction and getNextFrame() call.

However, I think it should be enough to do it just once in Msharpen::configure(void)

```
@@ -535,6 +544,7 @@ uint8_t r=0;
     if(DIA_msharpen(copy,this->previousFilter))
     {
         _param=copy;
+        invstrength=(_param.strength < 255)? 255-_param.strength : 0;
         return true;
     }
     return false;
```
rather than at every getNextFrame() invocation.